client/src/components/SignIn.jsx

```diff
@@ -34,11 +36,44 @@ function SignIn() {
       navigate('/dashboard');
     } catch (err) {
       setError(err.response?.data?.error || 'Something went wrong');
+    } finally {
+      setIsLoading(false);
     }
   };
 
+  const handleGoogleLogin = () => {
+    setIsGoogleLoading(true);
+  };
+
   return (
     <div className="signin-container">
+      {(isLoading || isGoogleLoading) && (
+        <div className="fancy-loading-overlay">
```

Comment:
can you consolidate this part to a new UI component like <LoadingScreen> something. I see both this and the other file uses it.